source/Messages.cpp

```diff
@@ -124,3 +131,28 @@ void Messages::Reset()
 	recent.clear();
 	logged.clear();
 }
+
+
+
+void Messages::LoadLog(const DataNode &node)
+{
+	for(const DataNode &child : node)
+	{
+		if(child.Size() < 2)
+			child.PrintTrace("Skipping message log entry without category:");
+		logged.emplace_front(child.Token(1), GameData::MessageCategories().Get(child.Token(0)));
+	}
+}
+
+
+
+void Messages::SaveLog(DataWriter &out)
+{
+	out.Write("message log");
+	out.BeginChild();
+	{
+		for(auto it = logged.rbegin(); it != logged.rend(); ++it)
+			out.Write(it->second->Name(), it->first);
```

Comment:
Why not this?
```suggestion
		for(auto &it : logged)
			out.Write(it.second->Name(), it.first);
```

Reply:
In the log, oldest messages are at the back, and I think it makes sense to save them chronologically.